pub_key.proto

```diff
@@ -24,26 +24,60 @@ message PubKeyMethod {
 }
 
 message NewPubKeyPayload {
-    enum EntityType {
-        PERSONAL = 0;
-        SERVER = 1;
+    enum PublicKeyType {
+        RSA = 0;
+        ECDSA = 1;
+        EDDSA = 2;
     }
 
-    enum PubKeyType {
-        RSA = 0;
+    enum HashingAlgorithm {
+        SHA256 = 0;
+        SHA512 = 1;
+    }
+
+    message RSAConfiguration {
+        enum Padding {
+            PSS = 0;
+            PKCS = 1;
+        }
+
+        uint64 modulus = 1;
+        bytes exponent = 2;
+        Padding padding = 3;
```

Comment:
maybe also oneof instead of Padding type that is enum?

Reply:
Arguments plz